tests/beacon/test_helpers.py

```diff
@@ -707,6 +722,228 @@ def test_get_domain(pre_fork_version,
+
+    votes = SlashableVoteData(**sample_slashable_vote_data_params)
+
+    keys = generate_aggregate_pubkeys(genesis_validators, votes)
+    assert len(keys) == 2
+
+    (poc_0_key, poc_1_key) = keys
+
+    poc_0_keys = get_pubkey_for_indices(genesis_validators, proof_of_custody_0_indices)
+    poc_1_keys = get_pubkey_for_indices(genesis_validators, proof_of_custody_1_indices)
+
+    assert bls.aggregate_pubkeys(poc_0_keys) == poc_0_key
+    assert bls.aggregate_pubkeys(poc_1_keys) == poc_1_key
+
+
+@given(st.data())
+def test_verify_vote_count(max_casper_votes, sample_slashable_vote_data_params, data):
+    (indices, some_index) = _list_and_index(data, max_size=max_casper_votes)
+    proof_of_custody_0_indices = indices[:some_index]
+    proof_of_custody_1_indices = indices[some_index:]
+
+    key = "aggregate_signature_poc_0_indices"
+    sample_slashable_vote_data_params[key] = proof_of_custody_0_indices
+    key = "aggregate_signature_poc_1_indices"
+    sample_slashable_vote_data_params[key] = proof_of_custody_1_indices
+
+    votes = SlashableVoteData(**sample_slashable_vote_data_params)
+
+    assert verify_vote_count(votes, max_casper_votes)
+
+
+def _get_indices_and_signatures(num_validators, message, privkeys):
+    num_indices = 5
+    assert num_validators >= num_indices
+    indices = random.sample(range(num_validators), num_indices)
+    privkeys = [privkeys[i] for i in indices]
+    domain = SignatureDomain.DOMAIN_ATTESTATION
+    signatures = tuple(
+        map(lambda key: bls.sign(message, key, domain), privkeys)
+    )
+    return (indices, signatures)
+
+
+def _correct_slashable_vote_data_params(params, validators, messages, privkeys):
+    valid_params = copy.deepcopy(params)
+
+    num_validators = len(validators)
+
+    key = "aggregate_signature_poc_0_indices"
+    (poc_0_indices, poc_0_signatures) = _get_indices_and_signatures(
+        num_validators,
+        messages[0],
+        privkeys,
+    )
+    valid_params[key] = poc_0_indices
+
+    key = "aggregate_signature_poc_1_indices"
+    # NOTE: does not guarantee non-empty intersection
+    (poc_1_indices, poc_1_signatures) = _get_indices_and_signatures(
+        num_validators,
+        messages[1],
+        privkeys,
+    )
+    valid_params[key] = poc_1_indices
+
+    signatures = poc_0_signatures + poc_1_signatures
+    aggregate_signature = bls.aggregate_signatures(signatures)
+
+    valid_params["aggregate_signature"] = aggregate_signature
+
+    return valid_params
+
+
+def _corrupt_signature(params):
+    params = copy.deepcopy(params)
+    message = bytes.fromhex("deadbeefcafe")
+    privkey = 42
+    domain = SignatureDomain.DOMAIN_ATTESTATION
+    params["aggregate_signature"] = bls.sign(message, privkey, domain)
+    return params
+
+
+def _corrupt_vote_count(params):
+    params = copy.deepcopy(params)
+    key = "aggregate_signature_poc_0_indices"
+    for i in itertools.count():
+        if i not in params[key]:
+            params[key].append(i)
```

Comment:
Here's how you can do this without the mutation.

```python
from toolz import assoc

def _corrupt_vote_count(params):
    for i in itertools.count():
        if i not in params[key]:
           return assoc(params, key, params[key] + [i])
    else:
        raise SomeException("Unreachable code path")
```

It's minor but it's nice to not mutate things.  If the `deepcopy` call somehow got dropped this would begin modifying the original `params` which isn't ideal.

Reply:
yeah this is nice and i agree safer than what was there w/ `copy.deepcopy`.